che-editors.yaml

```diff
@@ -1,373 +1,458 @@
-version: 1.0.0
+version: 2.0.0
 editors:
-  - id: eclipse/che-theia/next
-    title: Eclipse Theia development version.
-    displayName: theia-ide
-    description: Eclipse Theia, get the latest release each day.
-    icon: https://raw.githubusercontent.com/theia-ide/theia/master/logo/theia-logo-no-text-black.svg?sanitize=true
-    repository: https://github.com/eclipse-che/che-theia
-    firstPublicationDate: "2019-03-07"
-    endpoints:
-      - name: "theia"
-        public: true
-        targetPort: 3100
-        attributes:
-          protocol: http
-          type: ide
-          secure: true
-          cookiesAuthEnabled: true
-          discoverable: false
-      - name: "webviews"
-        public: true
-        targetPort: 3100
-        attributes:
-          protocol: http
-          type: webview
-          secure: true
-          cookiesAuthEnabled: true
-          discoverable: false
-          unique: true
-      - name: "mini-browser"
-        public: true
-        targetPort: 3100
-        attributes:
-          protocol: http
-          type: mini-browser
-          secure: true
-          cookiesAuthEnabled: true
-          discoverable: false
-          unique: true
-      - name: "theia-dev"
-        public: true
-        targetPort: 3130
-        attributes:
-          protocol: http
-          type: ide-dev
-          discoverable: false
-      - name: "theia-redirect-1"
-        public: true
-        targetPort: 13131
-        attributes:
-          protocol: http
-          discoverable: false
-      - name: "theia-redirect-2"
-        public: true
-        targetPort: 13132
-        attributes:
-          protocol: http
-          discoverable: false
-      - name: "theia-redirect-3"
-        public: true
-        targetPort: 13133
-        attributes:
-          protocol: http
-          discoverable: false
-      - name: terminal
-        public: true
-        targetPort: 3333
-        attributes:
-          protocol: ws
-          type: collocated-terminal
-          discoverable: false
-          secure: true
-          cookiesAuthEnabled: true
-    containers:
+  - schemaVersion: 2.1.0
+    metadata:
+      name: eclipse/che-theia/next
+      displayName: theia-ide
+      description: Eclipse Theia, get the latest release each day.
+      icon: https://raw.githubusercontent.com/theia-ide/theia/master/logo/theia-logo-no-text-black.svg?sanitize=true
+      attributes:
```

Comment:
based on https://github.com/devfile/api/pull/352#issuecomment-806857030

attributes in metadata are deprecated so probably need to move them to editors/attributes and using a freeForm format

as it's targeting plugin-registry only, maybe

```yaml
schemaVersion: 2.1.0
attributes:
  -pluginRegistry:
     - publisher: eclipse
     - version: next
     - repository: https://github.com/eclipse-che/che-theia
```

or we remove it from the definition of the devfile

```yaml
editors:
   - firstPublicationDate: 
      publisher: eclipse
      version: next
     repository: 
      editor:
        schemaVersion: 2.1.0
        components:....
```

 or we ask to not deprecate attributes in metadata (or we ask to have optional repository, firstPublicationDate, etc fields)

Reply:
According to https://github.com/devfile/api/pull/352#issuecomment-806857030 top level attributes are going to be added at the same time when the metadata attributes are deprecated. So I propose to leave it as it is but when the devfile API is updated, move the metadata attributes to top level attributes